python/fusion_engine_client/messages/configuration.py

```diff
@@ -399,3 +414,164 @@ def __str__(self):
 
     def calcsize(self) -> int:
         return len(self.pack())
+
+
+class InterfaceID(NamedTuple):
+    type: TransportType = TransportType.INVALID
+    index: int = 0
+
+
+class OutputInterfaceConfig(NamedTuple):
+    output_interface: InterfaceID = InterfaceID()
+    stream_indices: List[int] = []
+
+
+_InterfaceIDConstructRaw = Struct(
+    "type" / AutoEnum(Int8ul, TransportType),
+    "index" / Int8ul,
+    Padding(2)
+)
+_InterfaceIDConstruct = NamedTupleAdapter(InterfaceID, _InterfaceIDConstructRaw)
+
+
+class _OutputInterfaceConfigConstruct(Adapter):
+    """!
+    Adapter to handle setting `num_streams` implicitly.
+    """
+    def __init__(self):
+        super().__init__(Struct(
+            "output_interface" / _InterfaceIDConstruct,
+            "num_streams" / Int8ul,
+            Padding(3),
+            "stream_indices" / Array(this.num_streams, Int8ul),
+        ))
+
+    def _decode(self, obj, context, path):
+        return OutputInterfaceConfig(obj.output_interface, obj.stream_indices)
+
+    def _encode(self, obj, context, path):
+        return {
+            "output_interface": obj.output_interface,
+            "num_streams": len(obj.stream_indices),
+            "stream_indices": obj.stream_indices
+        }
+
+
+class SetOutputInterfaceStreamsMessage(MessagePayload):
+    """!
+    @brief Configure the set of output streams enabled for a given output interface.
+    """
+    MESSAGE_TYPE = MessageType.SET_OUTPUT_INFERFACE_STREAMS
+    MESSAGE_VERSION = 0
+
+    SetOutputInterfaceStreamsMessageConstruct = Struct(
+        "update_action" / AutoEnum(Int8ul, UpdateAction),
+        Padding(3),
+        "output_interface_config" / _OutputInterfaceConfigConstruct(),
+    )
+
+    def __init__(self):
+        self.update_action = UpdateAction.REPLACE
+        self.output_interface_config = OutputInterfaceConfig()
+
+    def pack(self, buffer: bytes = None, offset: int = 0, return_buffer: bool = True) -> (bytes, int):
+        packed_data = self.SetOutputInterfaceStreamsMessageConstruct.build(self.__dict__)
+        return PackedDataToBuffer(packed_data, buffer, offset, return_buffer)
+
+    def unpack(self, buffer: bytes, offset: int = 0) -> int:
+        parsed = self.SetOutputInterfaceStreamsMessageConstruct.parse(buffer[offset:])
+        self.__dict__.update(parsed)
+        return parsed._io.tell()
+
+    def __str__(self):
+        fields = ['update_action', 'output_interface_config']
+        string = f'Set Output Interface Streams Command\n'
+        for field in fields:
+            val = str(self.__dict__[field]).replace('Container:', '')
+            string += f'  {field}: {val}\n'
+        return string.rstrip()
+
+    def calcsize(self) -> int:
+        return len(self.pack())
+
+
+class GetOutputInterfaceStreamsMessage(MessagePayload):
+    """!
+    @brief Query the value of the output interface stream configuration.
```

Comment:
Maybe for clarity "Query the set of message streams configured to be output by the device on a specified interface." I find putting the words interface and stream together makes it a little less clear which thing is which.